pkcs11/_pkcs11.pyx

```diff
@@ -1084,414 +1346,445 @@ class DomainParameters(types.DomainParameters):
         cdef CK_RV retval
 
         with nogil:
-            retval = _funclist.C_GenerateKeyPair(handle, mech_data, public_attr_data, public_attr_count, private_attr_data, private_attr_count, &public_key, &private_key)
+            retval = session.funclist.C_GenerateKeyPair(session.handle, mech_data, public_attr_data, public_attr_count, private_attr_data, private_attr_count, &public_key, &private_key)
         assertRV(retval)
 
-        return (Object._make(self.session, public_key),
-                Object._make(self.session, private_key))
+        return (make_object(session, public_key),
+                make_object(session, private_key))
 
 
-class Certificate(types.Certificate):
+class LocalDomainParameters(GenerateWithParametersMixin, types.LocalDomainParameters):
     pass
 
+class StoredDomainParameters(GenerateWithParametersMixin):
+    pass
 
-class EncryptMixin(types.EncryptMixin):
-    """Expand EncryptMixin with an implementation."""
+class Certificate(types.Certificate):
+    pass
 
-    def _encrypt(self, data,
-                 mechanism=None, mechanism_param=None):
+
+cdef class KeyOperation(OperationContext):
```

Comment:
This context manager deals with cryptographic operations and the associated buffer management etc.